### PR TITLE
rn: add ability to set the conference subject

### PR DESCRIPTION
**Pull request by saghul**
Dear reviewer, see each individual commit for details.

### CODE REVIEW

File: android/sdk/src/main/java/org/jitsi/meet/sdk/JitsiMeetConferenceOptions.java [hunk 1]
```diff
@@ -105,6 +110,17 @@ public Builder setRoom(String room) {
             return this;
         }
 
+        /**
+         * Sets the conference title.
```
Comment (zbettenbuk) [inline android/sdk/src/main/java/org/jitsi/meet/sdk/JitsiMeetConferenceOptions.java]: Don't you wanna call it "conference subject" instead?

File: react/features/base/conference/actions.js [hunk 2]
```diff
@@ -757,10 +759,6 @@ export function setSubject(subject: string = '') {
         const { conference } = getState()['features/base/conference'];
 
         if (conference) {
-            dispatch({
-                type: SET_PENDING_SUBJECT_CHANGE,
```
Comment (zbettenbuk) [inline react/features/base/conference/actions.js]: What was this event originally?

File: react/features/base/conference/middleware.js [hunk 3]
```diff
@@ -324,10 +329,17 @@ function _connectionFailed({ dispatch, getState }, next, action) {
...
  */
-function _conferenceSubjectChanged({ getState }, next, action) {
+function _conferenceSubjectChanged({ dispatch, getState }, next, action) {
     const result = next(action);
     const { subject } = getState()['features/base/conference'];
 
+    if (subject) {
+        dispatch({
+            type: SET_PENDING_SUBJECT_CHANGE,
+            subject: undefined
```
Comment (zbettenbuk) [inline react/features/base/conference/middleware.js]: Your answer to my previous question may answer this as well, but why undefined if we check for subject previously?

File: android/sdk/src/main/java/org/jitsi/meet/sdk/JitsiMeetConferenceOptions.java (hunk 1 above)
Comment (saghul) [reply android/sdk/src/main/java/org/jitsi/meet/sdk/JitsiMeetConferenceOptions.java]: indeed

File: react/features/base/conference/actions.js (hunk 2 above)
Comment (saghul) [reply react/features/base/conference/actions.js]: it clears the "pending" subject, I moved that logic to the middleware, so it happens when the subject was updated

File: react/features/base/conference/middleware.js (hunk 3 above)
Comment (saghul) [reply react/features/base/conference/middleware.js]: this is the pending one, so after setting it, there is no pending subject anymore

**Pull request merged**